hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieWriterUtils.scala

```diff
@@ -179,9 +179,11 @@ object HoodieWriterUtils {
       if (null != tableConfig) {
         val datasourceRecordKey = params.getOrElse(RECORDKEY_FIELD.key(), null)
         val tableConfigRecordKey = tableConfig.getString(HoodieTableConfig.RECORDKEY_FIELDS)
-        if ((null != datasourceRecordKey && null != tableConfigRecordKey
-          && datasourceRecordKey != tableConfigRecordKey) || (null != datasourceRecordKey && datasourceRecordKey.nonEmpty
-          && tableConfigRecordKey == null)) {
+        val dsnull = datasourceRecordKey == null
+        val tcnull = tableConfigRecordKey == null
+        if ((!dsnull && !tcnull && datasourceRecordKey != tableConfigRecordKey)
+          || (!dsnull && datasourceRecordKey.nonEmpty
+          && tcnull) || ((dsnull || datasourceRecordKey.isEmpty) && !tcnull)) {
```

Comment:
I'm not sure, but I wonder if tableConfigRecordKey could be empty string
```suggestion
          && tcnull) || ((dsnull || datasourceRecordKey.isEmpty) && !tcnull && tableConfigRecordKey.nonEmpty)) {
```

Reply:
good suggestion